### PR TITLE
Fix account traits reducer

### PR DESCRIPTION
**Pull request by michaloo**

### CODE REVIEW

File: server/lib/map-filter-results.js [hunk 1]
```diff
@@ -44,3 +44,19 @@ export function reducePayload(payloadList: Array<Object>, byKey: string) {
     return acc;
   }, []);
 }
+
+
+export function reduceAccountPayload(payloadList: Array<Object>, byKey: string) {
+  return _.reduce(payloadList, (acc, payload) => {
+    if (!_.filter(acc, alreadyFiltered => _.isEqual(payload.accountIdentity, alreadyFiltered.accountIdentity)).length) {
```
Comment (unity) [inline server/lib/map-filter-results.js]: it seems that `byKey` here can be either `userIdentity` or `accountIdentity`. If so, could we write something like : 

```
 if (!_.filter(acc, alreadyFiltered => _.isEqual(payload[byKey], alreadyFiltered[byKey])).length) {
  const allOccurences = _.filter(payloadList, pld => _.isEqual(pld[byKey], payload[byKey]));
```
and have a single `reducePayload` method ?

Comment (michaloo) [reply server/lib/map-filter-results.js]: `byKey` is either `userTraits`, `accountTraits` or `accountIdent` but only for account links case.
To refactor it we would need to introduce two separate `byKey` like params. If doing so I would also change the name of the function to call it more like `deduplicate` or `unique` - reduce is rather generic term not telling what is the main purpose.

Comment (michaloo) [reply server/lib/map-filter-results.js]: I decided to go the shortest path possible knowing that this codebase is a no go longterm and it needs some refactor.

Comment (unity) [reply server/lib/map-filter-results.js]: ok. got it. if you're aware of this then fine by me.

**Pull request merged**